### PR TITLE
Feat/ubuntu to centos7

### PR DESCRIPTION
**Pull request by maneta**
I have migrated the Docker to Centos7
Added a Makefile
Added a .dockerignore

Please review and let me know if there any issues. It should be compatible with the mesos system.

I´m pushing this image to the tag centos7-to-ocp on quay.io (https://quay.io/repository/3scale/echoapi?tab=tags)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+all: build
+
+build: ## Build docker image with name LOCAL_IMAGE (NAME:VERSION).
+	docker build -f $(THISDIR_PATH)/Dockerfile -t $(LOCAL_IMAGE) $(PROJECT_PATH)
+
+test: ## Test built LOCAL_IMAGE (NAME:VERSION). 
+	docker run -t --env RACK_ENV=$(ENVIRONMENT) $(LOCAL_IMAGE) rackup -D
+	docker run -t $(LOCAL_IMAGE) 3scale_backend --version
```
Comment (davidor) [inline Makefile]: Looks like some of these tests do not belong here :)

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
+MAINTAINER Daniel Cesario <dcesario@redhat.com>
 
-RUN apt-install -y libssl-dev
+USER root
+
+RUN yum -y update \
+	&& yum install -y centos-release-scl \
+	&& yum install -y rh-ruby23 rh-ruby23-ruby-devel \
+	&& yum -y install make gcc \
+	&& yum -y install openssl-devel
```
Comment (mikz) [inline Dockerfile]: It is good to do `yum clean all` after install to clean the package cache

File: Makefile (hunk 1 above)
Comment (maneta) [inline Makefile]: Yes it is a wrong copy and paste, I actually make it right on 081f95b. Do you want I rebase the commits to get it out?

**Comment by mikz** [comment]
While it is a very good excercise to make own CentOS image with SCL and Ruby have you considered using `centos/ruby-23-centos7` ? That would have everything embedded and would be a two liner.

File: Dockerfile (hunk 2 above)
Comment (maneta) [inline Dockerfile]: Yes you are right, I was thinking on do it but I completely forgot to do so. My Bad I will update the PR.

**Comment by maneta** [comment]
Yes @mikz my first approach was to use the  "centos/ruby-23-centos7" but, for some reason, there was an unexpected exposed socket on port 8080. I commented it with @jmprusi, and we come on an agreement that start from a clean centos7 image would be more controlled than use the "centos/ruby-23-centos7".

File: Makefile (hunk 1 above)
Comment (davidor) [inline Makefile]: Yes :+1: I would do that to have a clean history.

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#Ignore documentation files
+	*.md
+
+#Igonre License
```
Comment (davidor) [inline .dockerignore]: typo

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	@sleep 1 
+	curl localhost:9292
+	docker kill $(VERSION) 
+	docker rm $(VERSION)
+
+run: ## Run the docker in the local machine.
+	docker run -t -P $(LOCAL_IMAGE)
+
+bash: ## Start bash in the build IMAGE_NAME.
+	docker run --entrypoint=/bin/bash -it $(LOCAL_IMAGE)
```
Comment (davidor) [inline Makefile]: How about adding `--rm` ?

File: Makefile (hunk 1 above)
Comment (maneta) [inline Makefile]: Ok no problemo 👍

File: Makefile (hunk 4 above)
Comment (maneta) [inline Makefile]: Of course 👍

File: Dockerfile [hunk 5]
```diff
@@ -1,18 +1,37 @@
...
 
 WORKDIR /opt/echo-api/
 
 COPY ./ /opt/echo-api
+COPY contrib/scl_enable /opt/echo-api/etc/
+
+RUN chown -fR 1001:1001 /opt/echo-api
+
+RUN	source /opt/echo-api/etc/scl_enable \
+	&& gem install -N bundler \
```
Comment (mikz) [inline Dockerfile]: always good to install a specific version with `--version=X`
the latest is `1.14.3` IIRC

**Pull request closed without merging**